src/libNode/ViewChangeBlockProcessing.cpp

```diff
@@ -238,14 +210,20 @@ bool Node::ProcessVCBlockCore(const VCBlock& vcblock) {
     return false;
   }
 
-  {
-    lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
-    for (unsigned int x = 0; x < newCandidateLeader; x++) {
-      // TODO: If VC select a random
-      // leader, we need to change the way
-      // we update ds composition.
-      UpdateDSCommiteeCompositionAfterVC(*m_mediator.m_DSCommittee);
+  for (const auto& faultyLeader : vcblock.GetHeader().GetFaultyLeaders()) {
```

Comment:
Hey, can you make this a separate function as I need it in the Directory Block verification

Reply:
Will extract to a new function then

Reply:
Thanks

Reply:
Implemented